gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -45,6 +45,30 @@ struct PatternBinding
   {}
 };
 
+// Info that gets stored in the map. Helps us detect if two bindings to the same
+// identifier have different mutability or ref states.
+struct BindingInfo
```

Comment:
I could see this struct becoming more complex I think you should make this a class and make these fields private and give them getters.